@navikt/core/css/darkside/typography.darkside.css

```diff
@@ -203,7 +203,16 @@
 }
 
 .navds-error-message {
+  display: flex;
```

Comment:
Display flex, gap and svg change should probably be "opt-in", so only when you set `icon` to true

Reply:
I'm not sure i follow, it should look fine with & without the new prop (so flex should look the same with 1 vs 2 child items)

Reply:
In most cases yes, but some teams can have used this component differently. So adding flex could (in some edge-cases) break something on their side.

Reply:
btw, most of it doesn't seem to be that new CSS either, (since I've moved that CSS from `form/form.darkside.css` to `typography.darkside.css`, in that file `.navds-error-message` was already flex with gap. The "new" CSS is mainly the svg selector, but there shouldn't be any svg's inside error messages currently right? 🤔 🤞 (i'll do a small check of the 227 uses of ErrorMessage 😅, maybe one of them added SVG's to the children)

Reply:
![image](https://github.com/user-attachments/assets/fb7a0285-5a6e-4ae9-828e-e1d2e9db3a49)
typography component should mean children === text, but the type is ReactNode 😅 (so technically anything could happen)

Reply:
Flex was previously only added when used inside one of our form field-components, our fieldset component or our file-item component:
```css
.navds-form-field .navds-error-message,
.navds-fieldset .navds-error-message,
.navds-file-item .navds-error-message {
  display: flex;
  gap: var(--a-spacing-2);
}
```

Reply:
ah, maybe there's a lot of ErrorMessage usage outside of form form components 🤔

Reply:
https://metabase.ansatt.nav.no/question/2711-radata-fra-component-usage?component=ErrorMessage&prop=&major=

There is a few uses it looks like
<img width="162" alt="Screenshot 2024-12-19 at 11 56 21" src="https://github.com/user-attachments/assets/4461729f-be2d-41a2-a2a9-21974fef3a2d" />

Reply:
Suddenly this became a near impossible thing to check 😅 ... it's now less effort to make it toggle off flex & gap when icon isn't set (it should probably retain color red though... and I think that's about it).

Reply:
> it's now less effort to make it toggle off flex & gap when icon isn't set (it should probably retain color red though... and I think that's about it).

Im a fan of the "think positive" approach Halvor has mentioned previously, so instead of a "navds-errormessage--no-icon" className, its instead "navds-errormessage--show-icon"

Reply:
yeah I was thinking to even use the positive data prop state convention now too:

(not final, but something like this)
```css
.navds-error-message[data-show-icon=true] {
  display: flex;
  gap: var(--a-spacing-1);
}

.navds-error-message[data-show-icon=true] svg {
  margin-top: 0.15em;
  flex: 0 0 auto;
  width: 1em;
  height: 1em;
}
```

Reply:
Me and Ken discussed the use of data-attributes the other day, and decided (at least for the time being) to only use data-attributes for states (like open/closed etc.)

Reply:
sounds good to me 👍, will then use the usual clsx BEM modifier instead (but ofc positive variant in the 99% of cases that works out best).